### PR TITLE
Edit workflows

### PR DESCRIPTION
**Pull request by itsacoyote**
# Description

Remove link-checker from Checks workflow. Removes the need to include a secret environment variable and build in the checks workflow. SEO Nuxt module is already configured to run a link checker on build.

Add environment tag to workflow for staging to track deployments.

Add codeowners to project. Requires DevXP team as PR reviewers on changes to project except for changes to /content and /public

### CODE REVIEW

File: .github/CODEOWNERS
Comment (MexicanAce) [inline .github/CODEOWNERS:7]: Do these specific directories need to have the GitHub Team specified? Or is it okay to leave them blank like that?

File: .github/CODEOWNERS
Comment (MexicanAce) [inline .github/CODEOWNERS:6]: Should we expand this to also include `txfusion`?
```suggestion
* @zksync-sdk/devxp @zksync-sdk/txfusion 
```

Comment (itsacoyote) [reply .github/CODEOWNERS:6]: Wasn't sure if they owned all of the docs as well to require them as part of the reviewers for `*`.

File: .github/CODEOWNERS
Comment (itsacoyote) [reply .github/CODEOWNERS:7]: The ruleset here is that devxp are owners for "everything" except for `/content` and `/public`. This way we ensure we review changes to the project structure but aren't required on the content and public folders so we don't hold up reviews in there. 

If txFusion should be owners for all the content in the /content directory I can also specify that.

File: .github/CODEOWNERS
Comment (MexicanAce) [reply .github/CODEOWNERS:6]: Ah, based on the other comment, limiting docs/repo structure changes can go through `devxp` for now

File: .github/CODEOWNERS
Comment (MexicanAce) [reply .github/CODEOWNERS:7]: Agreed about the split, and also think it's a good idea for both `devxp` and `txfusion` to be listed as reviewers for the `/content` directory.

Can we also add comments to the `CODEOWNERS` file with your explanation above?

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**
